lib/omnibus/fetchers/net_fetcher.rb

```diff
@@ -125,6 +125,11 @@ def checksum
       source[digest_type]
     end
 
+    # Set to false to extract archives into the root source_dir (default).
+    # Set to true to always extract underneath project_dir
+    #
+    attr_accessor :isolate_extraction
```

Comment:
I'm having the same thoughts as @schisamo 
I think this can be implemented in a fully backwards compatible way without an extra config option (I could be wrong, I dont know all the omnibus things):

Everything should start extracting into a folder exclusive for that package. Whenever builder steps are executed, they should be with cwd `source_dir/package_path/relative_path`. For this windows stuff, we dont specify relative path, and it's just `source_dir/package_path`. `source_dir` is something liek C:\omnibus-src\